src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/SubscriptionRouter.cs

```diff
@@ -15,19 +14,11 @@ public SubscriptionRouter(Publishers publishers, EndpointInstances endpointInsta
             this.transportAddressTranslation = transportAddressTranslation;
         }
 
-        public async Task<IEnumerable<string>> GetAddressesForEventType(Type messageType)
+        public IEnumerable<string> GetAddressesForEventType(Type messageType)
         {
-            var results = new HashSet<string>();
-            foreach (var publisherAddress in publishers.GetPublisherFor(messageType))
-            {
-                var addresses = await publisherAddress.Resolve(endpoint => endpointInstances.FindInstances(endpoint), i => transportAddressTranslation(i)).ConfigureAwait(false);
-                foreach (var address in addresses)
-                {
-                    results.Add(address);
-                }
-            }
-
-            return results;
+            var publisher = publishers.GetPublisherFor(messageType);
```

Comment:
I'm pretty sure we supported registering multiple publishers for the same type before. In our routing inheritance scenarios we saw there there are scenarios for that. Shouldn't we keep that or did I miss something?